src/engine/evaluate-patient.ts

```diff
@@ -67,21 +79,30 @@ function processCQLCommon(patient: Bundle, cql: string): Promise<PatientData> {
         return convertBasicCQL(cql);
       }
     })
-    .then(elm => {
-      let elmResults: ElmResults = {
-        patientResults: {}
-      };
-      if (instanceOfElmObject(elm)) {
-        elmResults = executeElm(patient, elm.main, elm.libraries);
-      } else {
-        elmResults = executeElm(patient, elm);
-      }
+    .then(elm => processELMCommon(patientRecord, elm));
+}
 
-      // TODO - update pathwaysData to take multiple patients
-      const patientIds = Object.keys(elmResults.patientResults);
-      const patientData = elmResults.patientResults[patientIds[0]];
-      return patientData;
-    });
+/**
+ * Common logic to execute the given ELM against the given Patient.
+ * @param patientRecord - Patient's record as FHIR data
+ * @param elm - resulting ELM from converting the CQL in a pathway
+ * @return the raw, unprocessed patientResults
+ *         derived from executing the ELM against the given patient
+ */
+function processELMCommon(patientRecord: Bundle, elm: object): PatientData {
```

Comment:
because `processCQLCommon` returns a Promise, you can save a few lines of code in other places by returning a promise here too (plus maybe avoid any astonishment from users who see the similar names).

```suggestion
function processELMCommon(patientRecord: Bundle, elm: object): Promise<PatientData> {
```

e.g. the `evaluatePathwayCriteria` method could be condensed to 4 lines instead of 8 and be a hair DRYer:


```typescript
export function evaluatePathwayCriteria(
  patientRecord: Bundle,
  pathway: Pathway
): Promise<CriteriaResult> {
  const patientData: Promise<PatientData> = pathway?.elm?.criteria
    ? processELMCommon(patientRecord, pathway.elm.criteria)
    : extractCriteriaCQL(pathway).then(cql => processCQLCommon(patientRecord, cql));
  return patientData.then(patientData => criteriaData(pathway, patientData));
}
```

Similarly for `evaluatePatientoOnPathway`.